tests/Feature/BookTest.php

```diff
@@ -79,4 +79,41 @@ public function test_getAllBooks_success(): void
             })
         ;
     }
+
+    public function test_getAllBooks_genre(): void
+    {
+        Book::factory()->count(10)->create();
+
+        $response = $this->get('/api/books?genre');
+
+        $response->assertStatus(200)
+            ->assertJson(function (AssertableJson $json) {
+                $json->hasAll(['data', 'message'])
+                    ->has('data', 10, function (AssertableJson $json) {
```

Comment:
And then change 10 to 1, since each book has it's own genre in your test data, you should only get 1 result back